src/main/java/org/folio/rest/impl/PostOrdersHelper.java

```diff
@@ -45,6 +46,9 @@ public PostOrdersHelper(HttpClientInterface httpClient, Map<String, String> okap
 
   public CompletableFuture<CompositePurchaseOrder> createPurchaseOrder(CompositePurchaseOrder compPO) {
     CompletableFuture<CompositePurchaseOrder> future = new VertxCompletableFuture<>(ctx);
+    if (compPO.getWorkflowStatus() == OPEN) {
```

Comment:
Even if an Order comes in as open we first initially set it to PENDING and only after other process(creation of PO Lines, inventory) happens we transition to OPEN. So this logic must be moved to PutOrdersHelper -> openOrder() method where we are transitioning the order to open.